tests/test_waveplate_thorlabs_k10cr1.py

```diff
@@ -46,27 +64,43 @@ def mock_send_message_expect_reply(
 
             assert match_reply_callback(reply_message)
 
-    ustatus_message = AptMessage_MGMSG_MOT_GET_STATUSUPDATE(
-        chan_ident=ChanIdent(1),
-        destination=Address.HOST_CONTROLLER,
-        source=Address.GENERIC_USB,
-        enc_count=0,
-        position=0,
-        status=Status(INMOTIONCCW=True, INMOTIONCW=True, HOMED=True),
-    )
+    def dynamic_mock() -> Callable[..., Any]:
+        call_count = 0
 
-    connection.send_message_expect_reply.side_effect = [
-        ustatus_message,
-        mock_send_message_expect_reply,
-    ]
-    connection.tx_ordered_sender_awaiting_reply = Mock()
-    connection.tx_ordered_sender_awaiting_reply.is_set = Mock(return_value=True)
+        def side_effect(*args: Any, **kwargs: Any) -> Any:
+            nonlocal call_count
+            call_count += 1
+
+            if call_count == 1:
+                return ustatus_message
+            if call_count == 2:
+                return mock_send_message_expect_reply(*args, **kwargs)
+            raise RuntimeError("Unexpected call count")
```

Comment:
While this works, it looks unnecessarily complex from here.

Why not:

1. Change `mock_send_message_expect_reply` to handle status messages; or
2. Completely  remove `mock_send_message_expect_reply` and just assign a list to `send_message_expect_reply.side_effect`

?

You're asserting the message behavior below, so it doesn't seem like you're getting a lot of benefit from the asserts in `mock_send_message_expect_reply`.